scipy/_lib/_array_api.py

```diff
@@ -275,6 +276,9 @@ def _strict_check(actual, desired, xp, *,
         assert actual.dtype == desired.dtype, _msg
 
     if check_shape:
+        if is_dask(xp):
+            actual.compute_chunk_sizes()
+            desired.compute_chunk_sizes()
```

Comment:
This can be very, very bad performance wise. This should instead be changed in line with https://github.com/data-apis/array-api-compat/issues/225, with check_shape=True/False/None where None means "check if possible".

Reply:
For clarity: my observations on performance should not be treated as blocking for this PR. They can be remediated at a later time.

Reply:
I see that this is only used within the scope of unit tests. So it should be fine to remain like this.